### ISSUE TITLE
Workaround for std::make_shared bug in Xcode 7.2 or ealier

### ISSUE THREAD

**Issue by tatsuhiro-t** [nghttp2/nghttp2#670]
https://github.com/curl/curl/commit/152e5c2105aa6b4701923b11096fcc30284ea848#commitcomment-18802516

In Xcode 7.2 or earlier, `std::make_shared` does not perform value-initialization.  Given the struct below:

```cpp
struct Foo {
    int a;
};
```

And, Foo object created by `std::make_shared` has `a` field undefined:

```cpp
auto f = std::make_shared<Foo>();
// This fails with Xcode 7.2 or earlier
assert(0 == f->a);
```

For workaround, do `f = {}` after `std::make_shared` if `__APPLE__` is defined.

We observed same bug in clang-3.6 which came with Ubuntu 15.04.

**Comment by tatsuhiro-t** [comment]
Which Xcode version does homebrew use to compile binary package?

**Comment by tatsuhiro-t** [comment]
Fix committed via 1dabe43ff4c63c342f1341cba7ffa4ecb1a465d7
